client/driver/docker.go

```diff
@@ -76,6 +76,22 @@ func (d *DockerDriver) Fingerprint(cfg *config.Config, node *structs.Node) (bool
 func createHostConfig(task *structs.Task) *docker.HostConfig {
 	// hostConfig holds options for the docker container that are unique to this
 	// machine, such as resource limits and port mappings
+
+        mode, ok := task.Config["network_mode"]
+        if !ok || mode == "" {
+                // docker default
+                logger.Printf("[WARN] driver.docker: no mode specified for networking, using Docker default")
+                mode = "default"
+        }
+
+        // Ignore the container mode for now
+        switch mode {
+        case "default", "bridge", "none", "host":
+                logger.Printf("[DEBUG] driver.docker: using %s as network mode", mode)
+        default:
+                logger.Printf("[WARN] invalid setting for network mode %s, defaulting to bridge mode on docker0", mode)
+                mode = "default"
```

Comment:
We should either set the mode to "bridge", or change the message to say "using default network mode on docker0".